### PR TITLE
Dbex/61680 datadog rum

### PR DESCRIPTION
**Pull request by micahaspyr**
## Summary

-  We are adding datadog tracking for the 526EZ form application. The application has been configured to load for logged in users

## Testing done

-  This testing will be done in staging

## What areas of the site does it impact? 
 The 526EZ application

### Quality Assurance & Testing

- [ ] Testing to be done preferably in staging

### Error Handling

- [] Browser console contains no warnings or errors.

### Authentication

- [] Did you login to a local build and verify all authenticated routes work as expected with a test user

### CODE REVIEW

File: src/applications/check-in/package.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
     },
     "licence": "MIT",
     "dependencies": {
-        "@datadog/browser-rum": "^4.22.0",
+        "@datadog/browser-rum": "^4.45.0",
```
Comment (acrollet) [inline src/applications/check-in/package.json:9]: This should probably get moved from the check-in workspace to the main package.json since it's getting used by multiple applications now. Now is probably as good a time as any, what do you think @department-of-veterans-affairs/patient-check-in ?

Also it would make sense to me to do that work in a separate PR without the 526EZ changes...

**Review by acrollet — CHANGES_REQUESTED** [review]

File: src/applications/check-in/package.json (hunk 1 above)
Comment (brianseek) [reply src/applications/check-in/package.json:9]: that all makes sense to me

File: src/applications/disability-benefits/all-claims/Form526EZApp.jsx [hunk 2]
```diff
@@ -255,4 +256,20 @@ const mapStateToProps = state => ({
...
+    env: environment.vspEnvironment(),
+    sessionSampleRate: 10, // Set to 100 to test locally
+    sessionReplaySampleRate: 20, // Set to 100 to test locally
+    trackUserInteractions: true,
+    trackResources: true,
+    trackLongTasks: true,
+    defaultPrivacyLevel: 'mask-user-input',
+  });
+  datadogRum.startSessionReplayRecording();
+}
```
Comment (Mottie) [inline src/applications/disability-benefits/all-claims/Form526EZApp.jsx:274]: I'd recommend initializing RUM replay recording behind a feature toggle. There's a lot of PII that'll show up in the recorded sessions until you find and mask/hide it. I spent quite a bit of time searching for and filtering out the PII in our relatively short form.

Comment (acrollet) [reply src/applications/disability-benefits/all-claims/Form526EZApp.jsx:274]: Thanks @Mottie, this is great advice. I'm curious if you tried using `defaultPrivacyLevel: 'mask',` and if so if you still saw PII coming through?

Comment (Mottie) [reply src/applications/disability-benefits/all-claims/Form526EZApp.jsx:274]: Mask only works for form elements, I think. We have a lot of pages that display content, so the class name is needed to mask/hide that content

Comment (micahaspyr) [reply src/applications/disability-benefits/all-claims/Form526EZApp.jsx:274]: I went ahead and masked PII html data one by one, and not just input.  Just using 'mask' renders all text as XXX, example from datadog for reference.  
image

File: src/applications/check-in/package.json (hunk 1 above)
Comment (micahaspyr) [reply src/applications/check-in/package.json:9]: I'll look into opening a PR during work hours for the version change.  For sure it's strange that's where the package was...

File: cypress.config.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+const { defineConfig } = require('cypress');
+
+module.exports = defineConfig({
+  e2e: {
+    // eslint-disable-next-line no-unused-vars
```
[the remark this answers is not in the document]
Comment (micahaspyr) [reply cypress.config.js]: Accidentally tracked this file

File: src/applications/check-in/package.json (hunk 1 above)
Comment (acrollet) [reply src/applications/check-in/package.json:9]: It was originally added to the check-in workspace because that was the only app using RUM initially.

File: src/applications/disability-benefits/all-claims/Form526EZApp.jsx (hunk 2 above)
Comment (acrollet) [reply src/applications/disability-benefits/all-claims/Form526EZApp.jsx:274]: Do you have a specific need to see certain parts unmasked? I just worry about the possibility of fields getting added later on and not masked.

File: src/applications/disability-benefits/all-claims/content/veteranDetails.jsx [hunk 4]
```diff
@@ -14,15 +14,15 @@ const unconnectedVetInfoView = profile => {
   return (
     <div>
       <p className="vads-u-margin-top--0">
-        This is the personal information we have on file for you.
+        This is the personal information we have on file for you sdfsdf.
```
Comment (Mottie) [inline src/applications/disability-benefits/all-claims/content/veteranDetails.jsx:17]: ??

File: src/applications/disability-benefits/all-claims/Form526EZApp.jsx (hunk 2 above)
Comment (Mottie) [reply src/applications/disability-benefits/all-claims/Form526EZApp.jsx:274]: From a quick glance at this PR, I think you're barely scratching the surface of all the PII - did you look at the review & submit page?

Comment (micahaspyr) [reply src/applications/disability-benefits/all-claims/Form526EZApp.jsx:274]: I was thinking we'd lose context by masking all text.  @va-albers @Mottie what do you think?

@Mottie I'll give it another run through for PII that may show

Comment (Mottie) [reply src/applications/disability-benefits/all-claims/Form526EZApp.jsx:274]: Sadly, you will lose context, but masking/hiding PII is important. Steve Albers had to check with platform security when I requested to keep the Country within the mailing address visible - it's not rendered when the Veteran selects United States; but it may help exposed issues with international addresses when they are used.

File: src/applications/disability-benefits/all-claims/content/veteranDetails.jsx (hunk 4 above)
Comment (tblackwe) [reply src/applications/disability-benefits/all-claims/content/veteranDetails.jsx:17]: @micahaspyr looks like some random characters got checked in, can clean it up

**Comment by micahaspyr** [comment]
@acrollet I'm getting code climate push back for what amounts to adding in the dd-privacy-mask into className.  Some of the code it complains about was forced upon me by the linter.  Is there a way to move past code climate as refactoring parts of the code climate issues are well beyong the scope of this PR.

**Comment by batemapf** [comment]
👋 we're working on a separate, supply-reordering-specific DD RUM [implementation over here](https://github.com/department-of-veterans-affairs/vets-website/pull/25656). would you be willing to pull your code changes out of the `.../2346/` namespace to reduce the possibility of conflicts?

**Comment by micahaspyr** [comment]
> 👋 we're working on a separate, supply-reordering-specific DD RUM [implementation over here](https://github.com/department-of-veterans-affairs/vets-website/pull/25656). would you be willing to pull your code changes out of the `.../2346/` namespace to reduce the possibility of conflicts?

@batemapf Yeah I can get that done by end of day today, if you can get me a reservation at Dorsia in return :)

**Comment by micahaspyr** [comment]
@batemapf I removed 'dd-privacy-mask' from files in the 2346 codespace for this PR.

**Comment by acrollet** [comment]
@micahaspyr the code climate checks are not blocking and you can use your own judgement about whether to ignore them. It makes a nice sanity check but is a bit over-eager to consider different blocks of code "identical", particularly in our monorepo situation.

File: src/applications/disability-benefits/all-claims/Form526EZApp.jsx [hunk 5]
```diff
@@ -270,4 +273,21 @@ const mapStateToProps = state => ({
   user: state.user,
 });
 
+if (isLoggedIn) {
```
Comment (acrollet) [inline src/applications/disability-benefits/all-claims/Form526EZApp.jsx:276]: I'd suggest adding a condition to make sure this doesn't run during CI, see [here](https://github.com/department-of-veterans-affairs/vets-website/blob/025469ecdf03146f5da04355523ee22e02acd205/src/applications/check-in/hooks/useDatadogRum.jsx#L11C5-L11C53) for an example...

**Review by acrollet — CHANGES_REQUESTED** [review]
